core/telemetry/impl/service_impl.cpp

```diff
@@ -257,7 +262,7 @@ namespace kagome::telemetry {
 
   void TelemetryServiceImpl::notifyBlockImported(
       const primitives::BlockInfo &info, BlockOrigin origin) {
-    if (shutdown_requested_) {
+    if (shutdown_requested_ or not enabled_) {
```

Comment:
```suggestion
    if (not enabled_ or shutdown_requested_) {
```
It is hint to compiler about **enabled_** is more actual condition.